server/auth/types/authentication_type.test.ts

```diff
@@ -106,4 +158,47 @@ describe('test tenant header', () => {
     const result = await dummyAuthType.authHandler(request, response, toolkit);
     expect(result.requestHeaders.securitytenant).toEqual('dummy_tenant');
   });
+
+  it(`keepalive should not shorten the cookie expiry`, async () => {
+    const keepAliveConfig = {
+      multitenancy: {
+        enabled: true,
+      },
+      auth: {
+        unauthenticated_routes: [] as string[],
+      },
+      session: {
+        keepalive: true,
+        ttl: 1000,
+      },
+    } as SecurityPluginConfigType;
+    const keepAliveDummyAuth = new DummyAuthType(
+      keepAliveConfig,
+      new BrowserSessionStorageFactory('security_cookie'),
+      router,
+      esClient,
+      coreSetup,
+      logger
+    );
+    const testCookie: SecuritySessionCookie = {
+      credentials: {
+        authHeaderValueExtra: true,
+      },
+      expiryTime: 2000,
+    };
+    // Set cookie
+    sessionStorage.setItem('security_cookie', JSON.stringify(testCookie));
+    const request = httpServerMock.createOpenSearchDashboardsRequest({
+      path: '/internal/v1',
+    });
+    const response = jest.fn();
+    const toolkit = {
+      authenticated: jest.fn((value) => value),
+    };
+    const _ = await keepAliveDummyAuth.authHandler(request, response, toolkit);
+    const cookieAfterRequest = sessionStorage.getItem('security_cookie');
+    expect(JSON.parse(cookieAfterRequest!).expiryTime).toBe(2000);
```

Comment:
What is this testing exactly? If keepalive is true and session.ttl is positive then shouldn't the test expect the expiry to be the current time + ttl?

The only usage I see of keepalive is here: https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/authentication_type.ts#L161-L165

Reply:
Good catch. I had some changes on my local I forgot to push. In this test, I am mocking date.now to return 0, setting the existing cookie expiry to 2000, and asserting that the cookie's expiry is not overwritten to a lower value (i.e. Date.now + ttl = 1000).